tracing.go

```diff
@@ -22,17 +22,31 @@ type tracingInfoKey struct{}
 
 // NewOpenTracingHooks provides a twirp.ServerHooks struct which records
 // OpenTracing spans.
-func NewOpenTracingHooks(tracer ot.Tracer) *twirp.ServerHooks {
+func NewOpenTracingHooks(tracer ot.Tracer, opts ...HookOpt) *twirp.ServerHooks {
 	hooks := &twirp.ServerHooks{
 		RequestReceived: startTraceSpan(tracer),
 		RequestRouted:   handleRequestRouted,
 		ResponseSent:    finishTrace,
-		Error:           handleError,
+		Error:           handleError(true),
+	}
+
+	for _, opt := range opts {
+		opt(hooks)
 	}
 
 	return hooks
 }
 
+type HookOpt func(h *twirp.ServerHooks)
```

Comment:
Hm, rather than naming this `HookOpt`, how do you feel about naming this `TraceServerOpt`? 

Also, it feels weird using functional options to override the entire `*twirp.ServerHooks` methods. I'm wondering if we can make a wrapper struct like this:

```go
type TracingServerHooks struct {
    opts *TraceServerOptions
}

type TraceServerOptions struct {
     IncludeClientErrors bool
     Tracer opentracing.Tracer
}

func IncludeClientErrors(includeClientErrors bool) TraceServerOpt {
        return func(opts *TraceServerOptions) {
               opts.IncludeClientErrors = includeClientErrors
        }
}

func NewOpenTracingHooks(tracer ot.Tracer, opts ...TraceServerOpt) *twirp.ServerHooks {
        defaultOpts := &TraceServerOptions{
              IncludeClientErrors: true,
              Tracer: tracer,
        }

        for _, opt := range opts {
             opt(defaultOpts)
        }

        traceHooks := &TraceServerHooks{
                    opts: defaultOpts,
         }
    
         return traceHooks.TwirpHooks()
}

func (t *TraceServerHooks) TwirpHooks() *twirp.ServerHooks {
        return &twirp.ServerHooks{
               RequestReceived: t.startTraceSpan(t.opts.tracer), // optionally this tracer can actually be left out since we have direct access to it
               RequestRouted: t.handleRequestRouted,
               ResponseSent: t.finishTrace,
               Error: t.handleError,
        }
}

func (t *TraceServerHooks) handleError(ctx context.Context, err twirp.Error) context.Context {
    // Reimplement `handleError` logic here.
    // ...
    if t.opts.IncludeClientErrors || statusCode >= 500 {
         span.SetTag("error", true)
    }
}
```
Or alternatively, we can just make the `opts` it's own struct and pass that into every single function that we have defined. What do you think?

Reply:
Done!  I didn't put the tracer in the opts since it's a hard requirement and options generally are optional, but everything else worked out smoothly.